client/ayon_resolve/plugins/create/create_workfile.py

```diff
@@ -52,9 +82,18 @@ def collect_instances(self):
                 False,
             )
         )
-        self.log.info("Auto-creating workfile instance...")
+
+        return data
+
+    def collect_instances(self):
+        """Collect from timeline marker or create a new one."""
+        data = self._loads_data_from_project_setting()
+        if not data:
+            self.log.info("Auto-creating workfile instance...")
+            data = self._create_new_instance()
+
         current_instance = CreatedInstance(
-            self.product_type, product_name, data, self)
+            self.product_type, data["productName"], data, self)
         self._add_instance_to_context(current_instance)
 
     def create(self, options=None):
```

Comment:
I vaguely recall @iLLiCiTiT you mentioning to me that there was _still_ a reason to implement `create` for `AutoCreator` instead of doing the 'creation' inside `collect_instances()` but I can't remember what it was. I just remember fighting you about it haha, but then subsiding because you had an explanation for it needing it.

Reply:
I remember there was a reason, now I don't remember (it's 2 years). I would say as simple as `collect_instances` should only collect existing instances (unless there is some backwards compatibility handling), `create` should create them and store the metadata to file.

Reply:
No problem. This is adjusted accordingly.